kythe/rust/indexer/src/providers.rs

```diff
@@ -56,12 +58,34 @@ impl KzipFileProvider {
     ///
     /// # Errors
     ///
-    /// If the file is not a valid zip archive, a
+    /// If an error occurs while reading the kzip, a
     /// [KzipFileError][KytheError::KzipFileError] will be returned.
     pub fn new(file: File) -> Result<Self, KytheError> {
         let reader = BufReader::new(file);
-        let zip_archive = ZipArchive::new(reader)?;
-        Ok(Self { zip_archive })
+        let mut zip_archive = ZipArchive::new(reader)?;
+
+        // Get the name of the root folder of the kzip. This should be almost always be
+        // "root" by the kzip spec doesn't guarantee it.
+        let root_name: String;
+        // Extra scrope is needed because we are borrowing zip_archive for file. File
+        // needs to get dropped so release the zip_archive borrow before we move
+        // it into a struct.
+        {
```

Comment:
I think you can collapse this as
```suggestion
        // Extra scrope is needed because we are borrowing zip_archive for file. File
        // needs to get dropped so release the zip_archive borrow before we move
        // it into a struct.
        let root_name = {
           // your impl
           path.to_str().unwrap().into()
        };
```

Reply:
Thanks for the suggestion! I forgot that Rust lets you do stuff like this, very cool.